consensus/view_change.go

```diff
@@ -468,22 +287,18 @@ func (consensus *Consensus) onViewChange(msg *msg_pb.Message) {
 					block.NumberU64(),
 					block.Header().ViewID().Uint64(),
 				); err != nil {
-					consensus.getLogger().Warn().Msg("submit vote on viewchange commit failed")
+					consensus.getLogger().Warn().Err(err).Msg("submit vote on viewchange commit failed")
 					return
 				}
 
 			}
 		}
 
-		consensus.SetViewChangingID(recvMsg.ViewID)
+		consensus.SetViewIDs(recvMsg.ViewID)
```

Comment:
can you add a todo here? // TODO: reconsider whether we should update CurBlockViewID yet because there is no new block confirmed yet and problem could still happen before that. It could cause problem if some nodes update their curBlockViewID locally while others don't, since view change now are based on time and CurBlockViewID.

Reply:
This makes sense, actually, the curblockviewID should be updated along with the commit of each block.  Let me add a TODO and dig deeper.